rollup/circuitcapacitychecker/libzkp/src/lib.rs

```diff
@@ -110,35 +79,41 @@ pub mod checker {
         serde_json::to_vec(&r).map_or(null(), vec_to_c_char)
     }
 
+    unsafe fn apply_tx_inner(id: u64, tx_traces: *const c_char) -> Result<RowUsage, Error> {
+        log::debug!(
+            "ccc apply_tx raw input, id: {:?}, tx_traces: {:?}",
+            id,
+            c_char_to_str(tx_traces)?
+        );
+        let tx_traces_vec = c_char_to_vec(tx_traces);
+        let traces = serde_json::from_slice::<BlockTrace>(&tx_traces_vec)?;
+
+        if traces.transactions.len() != 1 {
+            return Err(anyhow!("traces.transactions.len() != 1"));
```

Comment:
`bail!` is better, but it's OK.
```suggestion
            bail!("traces.transactions.len() != 1");
```